charts/nginx-gateway-fabric/templates/hpa.yaml

```diff
@@ -0,0 +1,46 @@
+{{- if and (eq .Values.nginxGateway.kind "deployment") .Values.nginxGateway.autoscaling.enabled -}}
+apiVersion: {{ ternary "autoscaling/v2" "autoscaling/v2beta2" (.Capabilities.APIVersions.Has "autoscaling/v2") }}
```

Comment:
Should `.Capabilities.APIVersions.Has "autoscaling/v2"` be a part of the `if` conditional above so that we aren't creating HPA if it can't exist?